fix(launcher): fix arguments handling in local launcher

sylvlecl opened the pull request:
- adds compatibility check for launcher arguments
- fixes presolve option handling for versions < 9.2
- fixes arguments formatting for subprocess: options and their arguments are separate command line arguments

Also fixes encoding-related errors when reading logs:
on windows in particular, error messages from the solver may have a locale-dependent encoding.
Not clear if the solver as a whole uses that same encoding, so we just assume utf-8 and replace
invalid characters with a replacement char...

sylvlecl committed:
fix(launcher): fix arguments handling in local launcher

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
to revert: build desktop

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Merge branch 'dev' into fix/fix-local-launcher-options

sylvlecl commented on .github/workflows/deploy.yml:
```diff
@@ -4,6 +4,7 @@ on:
     branches:
       - "master"
       - "hotfix/**"
+      - "fix/fix-local-launcher-options"
```
Remove it before merging

MartinBelthle replied:
As you said, should be removed

MartinBelthle commented on tests/launcher/test_local_launcher.py:
```diff
@@ -143,9 +142,9 @@ def test_parse_launcher_arguments(launcher_config: LocalConfig, xpress_env):
             sim_args, _ = local_launcher._parse_launcher_options(launcher_parameters, version)
             if version == solver_version_8_8:
-                assert sim_args == ["--use-ortools", f"--ortools-solver={solver}"]
+                assert sim_args == ["--use-ortools", "--ortools-solver", f"{solver}"]
```
Minor: I think f"{solver}" could be solver

MartinBelthle commented on tests/launcher/test_local_launcher.py:
```diff
@@ -166,14 +182,40 @@ def test_parse_launcher_arguments(launcher_config: LocalConfig, xpress_env):
         sim_args, _ = local_launcher._parse_launcher_options(launcher_parameters, version)
         arg = "lp" if version == solver_version_9_2 else "linear"
         assert sim_args == [
```
Not sure to see where you test this : `fixes arguments formatting for subprocess:` as the test seems to be the same as before

sylvlecl replied:
Tests have been adapted to check that arguments are correctly separated from their options, which is enough:
the problem was that `--linear-solver-param-optim-1="PRESOLVE 2 THREADS 4"` was provided as 1 argument, including the quotes. Unlike from a shell, the solver seemed unable to understand this.
Now we pass `--linear-solver-param-optim-1` and `PRESOLVE 2 THREADS 4` separately, without quotes, which fixes that issue.

MartinBelthle replied:
Okay I see

sylvlecl committed:
disable explicit encoding: we don't know the system configuration

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
remove redundant formatting

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
replace invalid utf-8 chars in logs

It's hard to be sure of the logs encoding, we go for the easy solution,
replacing invalid utf-8 characters.

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
remove desktop build

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
Merge remote-tracking branch 'origin/dev' into fix/fix-local-launcher-options
